### PR TITLE
fix(frontend): redirect after login

### LINKED ISSUE #1881: 🐛 [Bug] Redirect to deeplink after sign in
## 🐛 Bug
When I open a link to a table and I'm not logged in, I am redirected to Authentik, and after logging in, to the start page. I would like to be redirected to the table.
Same for all other links, or while surfing on the page and being logged out after a while of inactivity.

### PR DESCRIPTION
**Pull request by Bettelstab**
## 🍰 Pullrequest
When opening a link to e.g. `/table/5` in logged out state, now that path is stored as state of the oidc protocol and used to redirect the user there after successful sign in.

### Issues
- fixes #1881 

### Todo
Observe if it is necessary to pass the path in `apollo.ts` when backend sends `UNAUTHENTICATED`. (I did not implement it there because I could not think of a straight forward way to get `pageContext`.

### CODE REVIEW

**Review by roschaefer — CHANGES_REQUESTED** [review]
query params

File: frontend/src/services/AuthService.ts [hunk 1]
```diff
@@ -34,8 +34,12 @@ export default class AuthService {
     window.location.href = this.AUTH.AUTHORITY_SIGNUP_URI
   }
 
-  public signIn() {
-    return this.userManager.signinRedirect()
+  public signIn(pathname = '') {
+    return this.userManager.signinRedirect({
+      // State is sent back to the client after successfull login.
+      // We use the pathname as the state to redirect back to the original page.
+      state: `/${pathname}`,
```
[the remark this answers is not in the document]
Comment (Bettelstab) [reply frontend/src/services/AuthService.ts:41]: Not really, `state` is defined in the protocol. I could do something like:
`state: {
  redirectTo: '...',
}`
would you prefer that?

[the remark this answers is not in the document]
Comment (roschaefer) [reply frontend/src/services/AuthService.ts]: Yes, this is more telling.

File: frontend/src/pages/+guard.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ const guard: GuardSync = (pageContext): ReturnType<GuardSync> => {
   // there might be a better solution to combine both into pageContext
   const authStore = useAuthStore(pinia)
   if (!pageContext.hasToken && !authStore.isLoggedIn) {
-    throw redirect(`/signin${pageContext.urlPathname}`)
+    throw redirect(`/signin?previousUrl=${pageContext.urlPathname}`)
```
Comment (roschaefer) [inline frontend/src/pages/+guard.ts:13]: Watch out for encoding issues with slashes. You could e.g. use [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**